### DISCUSSION TITLE
Configurable option to pass -E to sudo

### DISCUSSION THREAD

**Discussion by alanzheng88** [deployphp/deployer#3903] [Help needed] [answered]
during the `deploy:update_code` task i need a way to run as `ubuntu` user while preserving the environment. was able to hack it by doing `set('become', 'ubuntu -E');` but was wondering if there's a better way to do this

**Comment by antonmedv** [comment, marked as answer]
> during the deploy:update_code task i need a way to run as ubuntu


Please elaborate little bit why?

> it by doing set('become', 'ubuntu -E');

It looks like an okay solution to me

**Reply by alanzheng88** [reply]
currently running git clone using credentials passed in through environment variables during the `deploy:update_code` task. when `deploy:update_code` task is executed, it's done with `sudo` privileges and by default doesnt inherit the environment. we also dont have ssh access for `deployer` user (despite it being created during provisioning step) so using `ubuntu` instead to clone the repo
